NitroxServer/Communication/Packets/Processors/ServerCommandProcessor.cs

```diff
@@ -1,23 +1,34 @@
 ﻿using NitroxModel.Packets;
 using NitroxServer.Communication.Packets.Processors.Abstract;
 using NitroxServer.ConsoleCommands.Processor;
+using NitroxServer.GameLogic.Players;
+using NitroxModel.DataStructures.GameLogic;
 
 namespace NitroxServer.Communication.Packets.Processors
 {
     public class ServerCommandProcessor : AuthenticatedPacketProcessor<ServerCommand>
     {
         private readonly ConsoleCommandProcessor cmdProcessor;
+        private readonly PlayerData playerData;
 
-        public ServerCommandProcessor(ConsoleCommandProcessor cmdProcessor)
+        public ServerCommandProcessor(ConsoleCommandProcessor cmdProcessor, PlayerData playerData)
         {
             this.cmdProcessor = cmdProcessor;
+            this.playerData = playerData;
         }
 
         public override void Process(ServerCommand packet, Player player)
         {
             string msg = string.Join(" ", packet.CmdArgs);
 
-            cmdProcessor.ProcessCommand(msg);
+            if (playerData.GetPermissions(player.Name) >= Perms.Admin)
```

Comment:
Maybe this should be handled outside of the actual command?  Commands can have a field like `MinPermissionRequired` and we don't even invoke if they don't meet the requirement.  That way you don't have to copy/paste this packet sending into every command.

Reply:
checked off the list